app/scripts/modules/netflix/pipeline/stage/isolatedTestingTarget/isolatedTestingTargetStage.ts

```diff
@@ -0,0 +1,252 @@
+import {module} from 'angular';
+import {filter, flatten, map, each} from 'lodash';
+import {IModalService} from 'angular-ui-bootstrap';
+
+import {AuthenticationService} from 'core/authentication/authentication.service';
+import {CLOUD_PROVIDER_REGISTRY, CloudProviderRegistry} from 'core/cloudProvider/cloudProvider.registry';
+import {ICluster, IStage, ServerGroup} from 'core/domain/index';
+import {SERVER_GROUP_COMMAND_BUILDER_SERVICE} from 'core/serverGroup/configure/common/serverGroupCommandBuilder.service';
+import {SERVER_GROUP_READER, ServerGroupReader} from 'core/serverGroup/serverGroupReader.service';
+import {NAMING_SERVICE, NamingService} from 'core/naming/naming.service';
+import {EDIT_VIP_MODAL_CONTROLLER} from './editVip.modal.controller';
+
+interface IVipOverride {
+  oldVip: string;
+  oldSecureVip: string;
+  newVip: string;
+  newSecureVip: string;
+  [key: string]: string;
```

Comment:
it may be worth adding a comment here explaining the purpose (since it's not really here to handle dynamic properties, which is usually why we add a field like this to an interface)